packages/transaction-controller/src/TransactionController.ts

```diff
@@ -547,6 +561,21 @@ export class TransactionController extends BaseController<
       if (!this.isHistoryDisabled) {
         addInitialHistorySnapshot(transactionMeta);
       }
+
+      // Check if swaps transaction
+      if (
+        [TransactionType.swap, TransactionType.swapApproval].includes(
+          transactionType,
+        ) &&
+        !this.isSwapsDisabled
+      ) {
+        transactionMeta = (await this.updateSwapsTransaction(
```

Comment:
Since the metadata hasn't yet been added to the state, is it simpler to remove all the return types here and let `updateSwapsTransaction` and the private methods update the metadata directly?

That way we wouldn't need to re-assign and the variable can remain `const`.

Reply:
Makes sense, changed the way to mutate object rather than returning it in https://github.com/MetaMask/core/pull/1877/commits/a5444f3e745e0d654a8a7874ac374197bd67e760